src/classes/typeValidation.ts

```diff
@@ -0,0 +1,134 @@
+//this class validates the MAIN input types
+export class Validation
```

Comment:
A lot of these validators are not doing a lot due to typescript already doing the type checking